src/Compiler/Microsoft.AspNetCore.Razor.Language/src/CodeGeneration/DefaultDocumentWriter.cs

```diff
@@ -78,16 +78,12 @@ public override void VisitDocument(DocumentIntermediateNode node)
                 // We only support algorithms that the debugger understands, which is currently SHA1 and SHA256.
 
                 string algorithmId;
-                var algorithm = Context.SourceDocument.GetChecksumAlgorithm();
-                if (string.Equals(algorithm, HashAlgorithmName.SHA256.Name, StringComparison.Ordinal))
+                var algorithm = Context.SourceDocument.SourceText.ChecksumAlgorithm;
+                if (algorithm == CodeAnalysis.Text.SourceHashAlgorithm.Sha256)
                 {
                     algorithmId = "{8829d00f-11b8-4213-878b-770e8597ac16}";
                 }
-                else if (string.Equals(algorithm, HashAlgorithmName.SHA1.Name, StringComparison.Ordinal) ||
-
-                    // In 2.0, we didn't actually expose the name of the algorithm, so it's possible we could get null here.
-                    // If that's the case, we just assume SHA1 since that's the only thing we supported in 2.0.
-                    algorithm == null)
+                else if (algorithm == CodeAnalysis.Text.SourceHashAlgorithm.Sha1)
```

Comment:
We no longer check for `algorithm == null`. Should we check for `== CodeAnalysis.Text.SourceHashAlgorithm.None` instead?

Reply:
If it's `None`, `SourceText.GetChecksum()` will throw an exception. No point in checking here.